File: base/src/main/java/material/com/base/BaseFragment.java
package material.com.base;

import android.os.Bundle;
import android.support.v4.app.Fragment;
/**
 * Created by zjl on 2017/3/27.
 */

public abstract class BaseFragment extends Fragment{
    private boolean HasLoadedOnce = false;
    public String sort;

    @Override
    public void setUserVisibleHint(boolean isVisibleToUser) {
        super.setUserVisibleHint(isVisibleToUser);
        Bundle bundle = getArguments();
        sort = bundle.getString("sort");
        if (isVisibleToUser){
            loadData();
        }
    }

    public abstract void loadData();
}

File: base/src/main/java/material/com/base/BaseFragment.kt
package material.com.base

import android.os.Bundle
import android.support.v4.app.Fragment

/**
 * Created by zjl on 2017/3/27.
 */

abstract class BaseFragment : Fragment() {
    private val HasLoadedOnce = false
    var sort: String?=null

    override fun setUserVisibleHint(isVisibleToUser: Boolean) {
        super.setUserVisibleHint(isVisibleToUser)
        val bundle = arguments
        sort = bundle.getString("sort")
        if (isVisibleToUser) {
            loadData()
        }
    }

    abstract fun loadData()
}
